### PR TITLE
new login logic with redirect and state fix

### PR DESCRIPTION
**Pull request by noahnovickf**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -7,19 +7,30 @@ import CreatePost from "containers/create-post";
 import Home from "containers/home";
 import Login from "containers/login";
 import ProtectedRoute from "containers/protected-route";
-import { addUserToState } from "store/thunks/users";
+import { addUserToState, handleNewUser } from "store/thunks/users";
```
Comment (lolcode8) [inline src/App.jsx:10]: `handleNewUser` isn't descriptive enough. Can change to something more readable. is it usersignup? or userlogin?

File: src/App.jsx [hunk 2]
```diff
@@ -7,19 +7,30 @@ import CreatePost from "containers/create-post";
...
 
   useEffect(() => {
     //on authstate changed
     const firebaseListener = firebase
       .auth()
       .onAuthStateChanged(function (user) {
-        if (user) {
-          props.addUserToStateThunk(user);
-        }
+        firebase
```
Comment (lolcode8) [inline src/App.jsx:21]: Sweet for now but we need to start using async await more

File: src/containers/login/login.connector.js [hunk 3]
```diff
@@ -2,8 +2,8 @@ import { connect } from "react-redux";
...
 
-const mapDispatchToProps = (dispatch) => ({
-  signInWithGoogleThunk: () => dispatch(signInWithGoogle()),
-});
+// const mapDispatchToProps = (dispatch) => ({
+//   signInWithGoogleThunk: () => dispatch(signInWithGoogle()),
+// });
 
-export default connect(null, mapDispatchToProps)(Login);
+export default connect(null, null)(Login);
```
Comment (lolcode8) [inline src/containers/login/login.connector.js:9]: This whole chunk can be now replaced with `export default Login`

File: src/App.jsx (hunk 1 above)
Comment (noahnovickf) [reply src/App.jsx:10]: did

File: src/containers/login/login.connector.js (hunk 3 above)
Comment (noahnovickf) [reply src/containers/login/login.connector.js:9]: Can't this just all be deleted?

**Pull request merged**